src/AppInstallerCommonCore/Experiment.cpp

```diff
@@ -0,0 +1,109 @@
+
+namespace AppInstaller::Settings
+{
+    namespace
+    {
+        ExperimentState GetExperimentStateInternal(Experiment::Key key, const UserSettings& userSettings)
+        {
+            if (key == Experiment::Key::None)
+            {
+                return { true, ExperimentToggleSource::Default };
+            }
+
+            if (!GroupPolicies().IsEnabled(TogglePolicy::Policy::Experiments))
+            {
+                AICLI_LOG(Core, Info, <<
+                    "Experiment " << Experiment::GetExperiment(key).Name() <<
+                    " is disabled due to group policy: " << TogglePolicy::GetPolicy(TogglePolicy::Policy::Experiments).RegValueName());
+                return { false, ExperimentToggleSource::Policy };
+            }
+
+            auto experiments = userSettings.Get<Setting::Experiments>();
+            auto experiment = Experiment::GetExperiment(key);
+            auto jsonName = std::string(experiment.JsonName());
+            if (experiments.find(jsonName) != experiments.end())
+            {
+                auto isEnabled = experiments[jsonName];
+                AICLI_LOG(Core, Info, <<
+                    "Experiment " << Experiment::GetExperiment(key).Name() <<
+                    " is set to " << isEnabled << " in user settings");
+                return { isEnabled, ExperimentToggleSource::UserSetting };
+            }
+
+            auto isEnabled = AppInstaller::Experiment::IsEnabled(experiment.GetKey());
+            AICLI_LOG(Core, Info, <<
+                "Experiment " << Experiment::GetExperiment(key).Name() <<
+                " is set to " << isEnabled);
+            return { isEnabled, ExperimentToggleSource::Default };
+        }
+
+        std::string ExperimentToggleSourceToString(ExperimentToggleSource source)
+        {
+            switch (source)
+            {
+            case ExperimentToggleSource::Default:
+                return "Default";
+            case ExperimentToggleSource::Policy:
+                return "Policy";
+            case ExperimentToggleSource::UserSetting:
+                return "UserSetting";
+            default:
+                return "Unknown";
+            }
+        }
+    }
+
+    std::string ExperimentState::ToJson() const
+    {
+        Json::Value root;
+        root["IsEnabled"] = m_isEnabled;
+        root["ToggleSource"] = ExperimentToggleSourceToString(m_toggleSource);
+        Json::StreamWriterBuilder builder;
+        return Json::writeString(builder, root);
+    }
+
+    ExperimentState Experiment::GetStateInternal(Key key)
+    {
+        return GetExperimentStateInternal(key, User());
+    }
+
+    ExperimentState Experiment::GetState(Key key)
+    {
+        return Logging::Telemetry().GetExperimentState(key);
+    }
+
+    Experiment Experiment::GetExperiment(Key key)
+    {
+        switch (key)
+        {
+        case Key::CDN:
+            return Experiment{ "CDN experiment", "CDN", "https://aka.ms/winget-settings", "CDN"};

```

Comment:
The first field looks like something that needs to be a loc string.

Reply:
I primarily used `ExperimentalFeature` as a [reference for this class's implementation](https://github.com/microsoft/winget-cli/blob/6034e7829616f838c691c1e2b10bc4cf6413a8ee/src/AppInstallerCommonCore/ExperimentalFeature.cpp#L77). But I can localize it if necessary.